{
  CancellationAwareTask task=new CancellationAwareTask();
  ExecutorService executor=Hazelcast.getExecutorService();
  Future future=executor.submit(task);
  assertTrue(future.cancel(true));
  assertTrue(future.isCancelled());
  try {
    future.get();
    fail("Should not complete the task successfully");
  }
 catch (  CancellationException e) {
  }
catch (  Exception e) {
    fail("Unexpected exception " + e.getMessage());
  }
}
